axelrod/fingerprint.py

```diff
@@ -0,0 +1,285 @@
+        new_data = [k.split(',') for k in data]
+        results = {}
+        for i in new_data:
+            edge = (int(i[0]), int(i[1]))
+            interactions = list(zip(i[4], i[5]))
+            if edge not in results:
+                results[edge] = []
+            results[edge].append(interactions)
+    return results
+
+
+class AshlockFingerprint():
+    def __init__(self, strategy, probe):
+        """
+        Parameters
+        ----------
+        strategy : class
+            A class that must be descended from axelrod.Player
+        probe : class
+            A class that must be descended from axelrod.Player
+        """
+        self.strategy = strategy
+        self.probe = probe
+
+    def construct_tournament_elements(self, step):
+        """Build the elements required for a spatial tournament
+
+        Parameters
+        ----------
+        step : float
+            The separation between each coordinate. Smaller steps will
+            produce more coordinates that will be closer together.
+
+        Returns
+        ----------
+        edges : list of tuples
+            A list containing tuples of length 2. All tuples will have either 0
+            or 1 as the first element. The second element is the index of the
+            corresponding probe (+2 to allow for including the Strategy and it's
+            Dual).
+
+        tournament_players : list
+            A list containing instances of axelrod.Player. The first item is the
+            original player, the second is the dual, the rest are the probes.
+
+        """
+        probe_coordinates = create_coordinates(step)
+        self.coordinates = probe_coordinates
+        edges = create_edges(probe_coordinates)
+
+        dual = DualTransformer()(self.strategy)()
+        probe_players = create_probes(self.probe, probe_coordinates)
+        # probes = probe_players.values()
+        tournament_players = [self.strategy(), dual] + probe_players
+
+        return edges, tournament_players
+
+    def fingerprint(self, turns=50, repetitions=10, step=0.01, processes=None):
+        """Build and play the spatial tournament.
+
+        Creates the probes and their edges then builds a spatial tournament
+        where the original strategy only plays probes whose coordinates sum to
+        less than 1 (or equal). Probes whose coordinates sum to more than 1 play
+        the Dual Strategy.
+
+        Parameters
+        ----------
+        turns : integer, optional
+            The number of turns per match
+        repetitions : integer, optional
+            The number of times the round robin should be repeated
+        step : float, optional
+            The separation between each coordinate. Smaller steps will
+            produce more coordinates that will be closer together.
+        processes : integer, optional
+            The number of processes to be used for parallel processing
+        """
+        outputfile = NamedTemporaryFile(mode='w')
```

Comment:
Needs test because this won't work on Windows I believe. 

Recall our conversation from last Monday: have the same options as for the `axelrod.Tournament.play` method: make it so that it can do things in memory if need be, similarly also allow for the option to write to a named file (because why not?). Then include a relevant check if on Windows (basically just do the same as the `Tournament` does).